src/game/engine/template.h

```diff
@@ -0,0 +1,58 @@
+/**
+ * @file
+ *
+ * @author tomsons26
+ *
+ * @brief
+ *
+ * @copyright Chronoshift is free software: you can redistribute it and/or
+ *            modify it under the terms of the GNU General Public License
+ *            as published by the Free Software Foundation, either version
+ *            2 of the License, or (at your option) any later version.
+ *            A full copy of the GNU General Public License can be found in
+ *            LICENSE
+ */
+#pragma once
+
+#ifndef TEMPLATE_H
+#define TEMPLATE_H
+
+#include "always.h"
+#include "object.h"
+#include "templatetype.h"
+
+#ifndef GAME_DLL
+TFixedIHeapClass<TemplateClass> g_Templates;
```

Comment:
This needs to be moved to the bottom of the file as TemplateClass is not yet defined.